### PR TITLE
Cookbook

### PR DESCRIPTION
**Pull request by darya-varabei**
- implement data loading from url using GDC
- loading image from url using NSOparationQueue
- improving the responsiveness of the app using multithreading

### CODE REVIEW

File: Cookbook/Cookbook/DetailViewController.swift [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    }
+}
+
+extension UIImageView {
+    public func imageFromUrl(urlString: String) {
+        if let url = NSURL(string: urlString) {
+            let request = NSURLRequest(url: url as URL)
+            NSURLConnection.sendAsynchronousRequest(request as URLRequest, queue: OperationQueue.main) {
+            (response: URLResponse?, data: Data?, error: Error?) -> Void in
+                self.image = UIImage(data: data! as Data)
```
[the remark this answers is not in the document]
Comment (EugeneItechart) [reply Cookbook/Cookbook/DetailViewController.swift:177]: crash operator is still there

File: Cookbook/Cookbook/DetailViewController.swift [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return CGFloat(TableConstants.rowHeight)
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let cell = Bundle.main.loadNibNamed("NutrientTableViewCell", owner: self, options: nil)?.first as? NutrientTableViewCell
+        cell?.nutrientType = self.nutrientsData[indexPath.row]
+         cell?.nutrientAmount = self.nutrientsData[indexPath.row + 4]
+        return cell!
```
Comment (EugeneItechart) [inline Cookbook/Cookbook/DetailViewController.swift:205]: force unwrapping. Why do you continue using it?

File: Cookbook/Cookbook/DetailViewController.swift [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    private func showSubviews() {
+        
+        let imageLoading = BlockOperation()
+        let titleLoading = BlockOperation()
+        let buttonLoading = BlockOperation()
+        let descriptionLoading = BlockOperation()
+        
+        imageLoading.addExecutionBlock {
+            DispatchQueue.main.async {
+                sleep(1)
```
Comment (EugeneItechart) [inline Cookbook/Cookbook/DetailViewController.swift:131]: Have you made any updates recently? We've discussed that sleep(1) is a bad practice especially on the main thread

File: Cookbook/Cookbook/DetailViewController.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (EugeneItechart) [reply Cookbook/Cookbook/DetailViewController.swift:177]: CRASH operator is still here!

**Review by EugeneItechart — CHANGES_REQUESTED** [review]
Please fix the comments and implement additional GCD use cases using disptach_group or dispatch_barrier

**Comment by darya-varabei** [comment]
Mentioned issues have been improved recently

**Review by EugeneItechart — APPROVED** [review]
Could be merged

**Pull request merged**